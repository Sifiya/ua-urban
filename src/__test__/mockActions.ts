import * as actions from '@/app/api/actions';
import * as authApi from '@/app/api/auth.api';

jest.mock('@/app/api/actions', () => ({
  addWordWithDefinition: jest.fn(),
  getAllWords: jest.fn(),
  addDefinition: jest.fn(),
}));

export const mockAddWordWithDefinition = actions.addWordWithDefinition as jest.Mock;
export const mockGetAllWords = actions.getAllWords as jest.Mock;
export const mockAddDefinition = actions.addDefinition as jest.Mock;

jest.mock('@/app/api/auth.api', () => ({
  getUser: jest.fn(),
}));

export const mockGetUser = authApi.getUser as jest.Mock;
