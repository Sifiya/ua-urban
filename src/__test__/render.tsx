import React, { ReactNode } from 'react';
import { render as libRender } from '@testing-library/react';
import { Providers } from '@/components/Providers';

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <Providers isTest>{children}</Providers>;
};

export const render = (ui: ReactNode) => {
  return libRender(ui, {
    wrapper: Wrapper,
  });
};
