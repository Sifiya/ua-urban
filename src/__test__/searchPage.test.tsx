import React from 'react';
import { screen } from '@testing-library/react';
import { render } from '@/__test__/render';
import SearchPage, { generateMetadata } from '../app/search/page';

describe('SearchPage', () => {
  describe('generateMetadata', () => {
    test('should return title', async () => {
      const metadata = await generateMetadata({ searchParams: { word: 'word' } });
      expect(metadata.title).toEqual('Шукати слово word');
    });
  });

  describe('SearchPage', () => {
    test('should render', async () => {
      const { container } = render(<SearchPage searchParams={{ word: '' } }/>);
      expect(container).toMatchSnapshot();
    });
  });
});
