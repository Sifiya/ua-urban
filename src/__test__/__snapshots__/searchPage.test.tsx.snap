// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage SearchPage should render 1`] = `
<div>
  <div
    class="flex flex-col items-center"
  >
    <div
      class="w-full -mb-5"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 px-4 py-2 -ml-4"
      >
        <svg
          fill="currentColor"
          height="16"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          />
        </svg>
        <span
          class="ml-1.5"
        >
          Назад
        </span>
      </button>
    </div>
    <div
      class="rounded-lg border shadow-sm w-full max-w-[700px] mt-8 mb-3 bg-primary text-primary-foreground"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="text-2xl font-semibold leading-none tracking-tight"
        >
          Пошук
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <div
          class="flex items-center gap-2 mb-3"
        >
          <svg
            class="text-muted/60"
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 text-foreground"
            placeholder="Шукати слово..."
            value=""
          />
          <a
            href="/search?word="
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary hover:bg-secondary/80 h-10 px-4 py-2 text-foreground"
            >
              Пошук
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="w-full max-w-[700px]"
    />
  </div>
</div>
`;
